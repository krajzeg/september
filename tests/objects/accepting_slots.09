# test adding fields
hitchhikersGuide := Object()
hitchhikersGuide.accept("theAnswer", Slot.field())
print("The manually added field should have nothing inside:", hitchhikersGuide.theAnswer)
hitchhikersGuide.theAnswer = 42
print("But we can set it (to 42, for example):", hitchhikersGuide.theAnswer)

# test adding methods
checkAnswer := |yourAnswer| { return (yourAnswer == theAnswer) }
hitchhikersGuide.accept("checkAnswer", Slot.method(checkAnswer))
print("The manually added method should yield true:", hitchhikersGuide.checkAnswer(42))
