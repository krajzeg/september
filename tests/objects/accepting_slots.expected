The manually added field should have nothing inside: <Nothing>
But we can set it (to 42, for example): 42
The manually added method should yield true: <True>
